code/script.js

```diff
@@ -262,82 +417,163 @@ const selectQuestion = () => {
+  if (keep) {
+    alert(
+      `Correctamundo, the person has ${value} eyes! Keep all that has ${value} eyes.`
+    )
+  } else {
+    alert(
+      `Sorry, the person doesn't have ${value} eyes. Remove all that has ${value} eyes.`
+      )
+  }
+} else if (category === 'other') {
+  if (keep) {
+    alert(
+      `Affirmative, the person has a ${attribute}. Keep all that has a ${attribute}.`
+    )
+} else {
+    alert(
+      `Negative, the person doesn't have a ${attribute}. Remove all that has a ${attribute}.`
+    )
+  } 
+}
   // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
+  
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
+  generateBoard()
+  addGuessesMade(keep)
   // Invoke a function to redraw the board with the remaining people.
 }
 
+// Adds the current guess to the list of guesses made
+const addGuessesMade = (keep) => {
+  if (keep) {
+    guessesMade.innerHTML += `
+  <span class="right-guess"><img class="yes-or-no" src="./images/yes.svg">${currentQuestion.text}</span>
+  `
+  }
+  else {
+    guessesMade.innerHTML += `
+    <span class="wrong-guess"><img class="yes-or-no" src="./images/no.svg">${currentQuestion.text}</span>
+    `
+  }
+}
+
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
+  const confirmed = confirm(`Do you think the secret person is ${suspect}?`)
+
+  if (confirmed) {
+    checkMyGuess(suspect)
+  }
+// store the interaction from the player in a variable.
   // remember the confirm() ?
   // If the player wants to guess, invoke the checkMyGuess function.
-}
+};
 
 // If you confirm, this function is invoked
 const checkMyGuess = (suspect) => {
-  // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+
+  if (suspect === secret.name) {
+    winOrLoseText.innerHTML = `You are a genius!! It was ${suspect}.`
+  } else {
+    winOrLoseText.innerHTML = `It is not ${suspect}. It was ${secret.name}.`
+  }
+  
+  // 1.Check if the suspect is the same as the secret person's name
+  // 2.Set a Message to show in the win or lose section accordingly
+  // 3.Show the win or lose section
+  // 4.Hide the game board
+  winOrLose.style.display = 'flex'
+  board.style.display = 'none'
 }
 
+playAgain.addEventListener('click', () => {
+  location.reload()
+})
+
+// Sets the timer
+setInterval(() => {
+  timePassed++
+  time.innerText = timePassed
+}, 1000)
```

Comment:
Love this! 😍 I was myself struggling with presenting the timer for my game, but you managed to solve it with such elegance! 🥇 
Great Idea to use setInterval()
🎉 🎉

Reply:
I had help! But yes, it turned out well.